### PR TITLE
test: raise coverage toward 90% (Phase 2 — pandas/file validators)

### PR DESCRIPTION
**Pull request by divyasinghds**
## Summary

Phase 2 of the 90% coverage effort (Phase 1 = #126, already merged). Adds tests for the pandas- and filesystem-backed validators.

- `data_validator` — every per-type validator (INT/BIGINT, FLOAT/DOUBLE/DECIMAL, VARCHAR/CHAR/TEXT, BOOLEAN across dtypes, DATE/DATETIME/TIMESTAMP/TIME), constraint stripping, and `_detect_column_type`.
- `xml_validator` — full Pascal VOC structure: root/source/size/object/bndbox sub-validators, parse errors, coordinate relationships, file discovery.
- `image_validator` — the `validate()` discovery + uniformity flow (resolution-matching unit tests already existed).
- `file_validator` — extension uniformity, strict-mode rejection, file discovery.
- `duplicate_validator` — parent-dir warning, dir creation, error fallbacks.

**Coverage: 53% → 73%.** Per module: file_validator 99%, image 91%, xml 90%, data_validator 87%, duplicate 85%.

## Phased plan (remaining)

- **Phase 3** — ingestors (`base`/`csv`/`json`), `file_transfer`, `api/client`, `validators_mapping`
- **Phase 4** — `database.py` via mocked SQLAlchemy

## Test plan

- [ ] CI green on Python 3.11 + 3.12

---

> [!NOTE]
> **Low Risk**
> Test-only changes with no runtime or ingest behavior modifications; risk is limited to CI maintenance if fixtures or env handling drift.
> 
> **Overview**
> Phase 2 of the coverage push adds **five new pytest modules** (~760 lines) that exercise pandas- and filesystem-backed validators; production code is unchanged.
> 
> **`DataValidator`** (`test_data_validator.py`): top-level `validate()` (no schema, empty data, CSV load, unknown types), per-SQL-type rules (INT/BIGINT, float family, VARCHAR/CHAR/TEXT, BOOLEAN variants, date family), constraint stripping on types, and `_detect_column_type`.
> 
> **`PascalVOCXMLValidator`** (`test_xml_validator.py`): directory flow under `SRC_PATH`, file discovery helpers, and granular VOC checks (root/source/size/object/bndbox, parse errors, warnings).
> 
> **`ImageResolutionValidator`** (`test_image_validator_flow.py`): `SRC_PATH/images` discovery, uniform vs mixed resolutions, expected size mismatch, and helper edge cases (complements existing resolution unit tests).
> 
> **`FileTypeValidator`** (`test_file_validator.py`): extension uniformity, strict mode, custom subpaths, and `_get_files_to_validate` behavior with `clean_env` + `tmp_path`.
> 
> **`DuplicateValidator`** (`test_duplicate_validator_extra.py`): parent-dir warnings, directory existence/empty checks, and `_create_directory_if_needed` paths not covered elsewhere.
> 
> Together these tests are intended to raise overall coverage from ~53% toward ~73% for the targeted validator modules.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 6e7fad14aef1b7e17236c149f037858c2e316a76. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: tests/test_xml_validator.py [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+@pytest.fixture
+def write_xml(tmp_path):
+    """Write XML text into SRC_PATH and point config at it via env."""
+
+    def _write(xml_text, *, name="ann.xml", monkeypatch=None):
+        path = tmp_path / name
+        path.write_text(xml_text, encoding="utf-8")
+        return path
+
+    return _write
```
[the remark this answers is not in the document]
Comment (divyasinghds) [reply tests/test_xml_validator.py]: Fixed in 6e7fad1 — removed the unused `write_xml` fixture; tests use direct writes or the module-level `_voc_xml()`/`_root()` helpers.

**Review by aptracebloc — APPROVED** [review]
